toolkit/livedata/src/test/kotlin/br/com/arch/toolkit/livedata/response/SwapResponseLiveDataTest.kt

```diff
@@ -0,0 +1,112 @@
+package br.com.arch.toolkit.livedata.response
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleOwner
+import android.arch.lifecycle.LifecycleRegistry
+import android.arch.lifecycle.LiveData
+import com.nhaarman.mockitokotlin2.mock
+import net.vidageek.mirror.dsl.Mirror
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+
+class SwapResponseLiveDataTest {
+
+    @Rule
+    @get:Rule
+    @JvmField
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private var owner = object : LifecycleOwner {
+        private val registry = LifecycleRegistry(this)
+        override fun getLifecycle(): Lifecycle {
+            registry.markState(Lifecycle.State.RESUMED)
+            return registry
+        }
+    }
+
+    @Test
+    fun whenSwapSource_withoutTransformation_shouldReplicateData() {
+        val mockedObserver: (String) -> Unit = mock()
+
+        val liveData = MutableResponseLiveData<String>()
+        val swapLiveData = SwapResponseLiveData<String>()
+        Assert.assertFalse(swapLiveData.hasDataSource)
+
+        swapLiveData.swapSource(liveData)
+        Assert.assertTrue(swapLiveData.hasDataSource)
+
+        val data = "data"
+        liveData.postData(data)
+        swapLiveData.observeData(owner, mockedObserver)
+        Mockito.verify(mockedObserver).invoke(data)
+    }
+
+    @Test
+    fun whenSwapSource_withoutTransformation_discardAfterLoading_shouldReplicateData_nullDataBefore() {
+        val mockedObserver: (String) -> Unit = mock()
+
+        val liveData = MutableResponseLiveData<String>()
+        val swapLiveData = SwapResponseLiveData<String>()
+        Assert.assertFalse(swapLiveData.hasDataSource)
+
+        swapLiveData.swapSource(liveData, true)
+        Assert.assertTrue(swapLiveData.hasDataSource)
+
+        val data = "data"
+        liveData.postData(data)
+        swapLiveData.observeSingleData(owner, mockedObserver)
+        Mockito.verify(mockedObserver).invoke(data)
+
+        Assert.assertNull(swapLiveData.value)
+
+        LiveData::class.java.declaredMethods.find { it.name == "postValue" }?.let {
+            it.isAccessible = true
+            it.invoke(liveData, null)
+        }
+        swapLiveData.observeSingleData(owner, mockedObserver)
+        Mockito.verify(mockedObserver).invoke(data)
+
+        Assert.assertNull(swapLiveData.value)
+    }
+
+    @Test
+    fun whenSwapSource_withTransformation_shouldReplicateTransformedData() {

```

Comment:
Thread.sleep is necessary because I'm breaking another Thread inside the method "swap"... So it was causing some delay in the execution of the test. Sorry =(

Reply:
And actually, Depending of the interpretation, I could break is more test methods, but, I'm testing the method with one set of parameters and his behavior. I think it is acceptable. xP